pkg/systray/runtime/runner.go

```diff
@@ -58,16 +62,42 @@ func (r *SystrayUsersProcessesRunner) Execute() error {
 // Interrupt stops creating launcher systray processes and kills any existing ones.
 func (r *SystrayUsersProcessesRunner) Interrupt(err error) {
 	r.interrupt <- struct{}{}
+
 	for _, proc := range r.uidProcs {
```

Comment:
I think this for loop feel a little weird, but I can't quite say why. I think it's probably fine to merge, and at worst will be noisy and we'll iterate.

Reply:
yea, it's weird ....

how about ...

```go
if !processExists(proc.Pid) {
	continue
} else if err := proc.Signal(syscall.SIGTERM); err != nil {
	level.Error(r.logger).Log(
		"msg", "error sending SIGTERM to systray process after interrupt",
		"err", err,
	)
} else if err := proc.Kill(); err != nil {
	level.Error(r.logger).Log(
		"msg", "error killing systray process after interrupt",
		"err", err,
	)
}
```

could use a switch statement, but then you can't capture the err in a variable for logging

```go
switch {
case !processExists(proc.Pid):
	continue
case proc.Signal(syscall.SIGTERM) == nil:
	continue
case proc.Kill() == nil:
	continue
default:
	level.Error(r.logger).Log(
		"msg", "sigterm and kill failed for process",
		"pid", proc.Pid,
	)
}
```

Reply:
Yeah, these are both getting at what I mean... Though Ideally you want a small sleep between the kills.

I almost want something like:
```
for _, proc := range r.uidProcs {
   proc.Signal(syscall.SIGTERM)
}

select {
case <-wgDone:
    return
case <-time.NewTimer(time.Second * 5).C:
    for _, proc := range r.uidProcs {
        proc.Kill()
    }
case <-time.NewTimer(time.Second * 5).C:
    level.Error(r.logger).Log("msg", "timeout waiting for systray processes to finish")
}
```

can't quite tell if that's correct, or too complicated, or what